### ISSUE TITLE
Does not destroy frame

### ISSUE THREAD

**Issue by Holly23i** [Holly23i/coursework#4]
https://github.com/Holly23i/coursework/blob/0054ba217fb2faba00fd63747e84be7b2bb04f2e/pages.py#L404

**Comment by WK-GiHu** [comment]
@Holly23i 

> I don't see `self.destroy()` in `pages.py` at Line 404?

**Comment by Holly23i** [comment]
It's a recent update, maybe refresh your page or follow the link above?

**Comment by WK-GiHu** [comment]
@Holly23i 

Yes, got it, see my edited comment

**Comment by Holly23i** [comment]
What I want to achieve is to remove all the widgets from that frame, I don't know how I can achieve that?

**Comment by WK-GiHu** [comment]
@Holly23i 

>I want to achieve is to remove all the widgets from that frame, I don't know how I can achieve that?

STEP 1: Remove the following 2 Lines (394, 395), its useless.

https://github.com/Holly23i/coursework/blob/0054ba217fb2faba00fd63747e84be7b2bb04f2e/pages.py#L394

Update `pages.py` to review STEP 2

**Comment by WK-GiHu** [comment]
@Holly23i 

### STEP 2:

- [ ] Add before
https://github.com/Holly23i/coursework/blob/47e5e12561687802a9bd89326544d4faf080338f/pages.py#L401

  the following:
  ```
    def switch_frame_and_destroy(self):
        self.master.master.showFrame(homePage)

        for widget in self.grid_slaves():
            widget.destroy()
  ```

- [ ] Delete this statement:
https://github.com/Holly23i/coursework/blob/47e5e12561687802a9bd89326544d4faf080338f/pages.py#L404


- [ ] Change command from,
https://github.com/Holly23i/coursework/blob/47e5e12561687802a9bd89326544d4faf080338f/pages.py#L443

  to:
  ```
  command=self.switch_frame_and_destroy()
  ```

**Comment by Holly23i** [comment]
@WK-GiHu 
I don't want that button to destroy the frame, I want it to destroy upon calling displayResults so I removed the `self.master.master.showFrame(homePage)`. Defined the function above `onShowFrame` inline with onShowFrame and it gave me the `NameError: name 'destroy_frame' is not defined` error. I renamed the function `destroy_frame`. I have run into this before when defining functions around the classes but I didn't find where to define the function.
